include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -99,4 +99,15 @@ def MemoryOp : CalyxPrimitive<"memory", [
       "ArrayRef<int64_t>":$addrSizes
     )>
   ];
+
+  let extraClassDeclaration = [{
+    size_t getNumAddresses() { return addrSizes().size(); }
+
+    ValueRange addrPorts() { return getResults().take_front(getNumAddresses()); }
```

Comment:
Why not index accessor for this as well?

Reply:
By this, do you mean a `Value addrPort(size_t i)` function?

Reply:
Yeah, as you did for `addr(size_t i)`. An accessor by index and a ValueRange accessor for both makes sense to me. When working on frontend stuff for Calyx, I recall using both of these.